File: lab-solutions/Lab7/RoomPersistenceLabSkeleton/app/src/main/java/mobiledev/unb/ca/roompersistencelab/ItemDao.java
package mobiledev.unb.ca.roompersistencelab;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

/**
 * This DAO object validates the SQL at compile-time and associates it with a method
 */
@Dao
public interface ItemDao {
    // TODO Add app specific queries in here
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Item item);

    @Query("SELECT * FROM item_table WHERE name = :name ORDER BY num")
    List<Item> findItemsWithName(String name);

    //@Query("SELECT * FROM item_table WHERE name = :name ORDER BY num")
    //LiveData<List<Item>> findItemsWithName(String name);
}

File: Fall2023/examples/Lecture 7/RoomPersistenceLibraryCoroutinesDemo/app/src/main/java/mobiledev/unb/ca/roomcoroutinetest/dao/ItemDao.kt
package mobiledev.unb.ca.roomcoroutinetest.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import mobiledev.unb.ca.roomcoroutinetest.entity.Item

/**
 * This DAO object validates the SQL at compile-time and associates it with a method
 */
@Dao
interface ItemDao {
    @Query("SELECT * from item_table ORDER BY id ASC")
    fun listAllRecords(): LiveData<List<Item>>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(item: Item)

    @Delete
    fun deleteItem(item: Item)
}
